sdcm/provision/common/region_fallback.py

```
@@ -0,0 +1,176 @@
+        except Exception as exc:  # noqa: BLE001
+            if not is_capacity_error(exc):
+                restore()
+                raise
+            dc_index = failed_dc_index(exc)
+            cleanup()
+            candidate = next_candidate(dc_index) if dc_index is not None else None
+            if candidate is None:
+                restore()
+                raise error_factory(
+                    f"Failed creating multi-DC clusters; DC {dc_index} exhausted its fallback candidates: {exc}"
+                )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)